components/email.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Manage BuddyPress Email Post Types.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Email extends BPCLI_Component {
+	/**
+	 * Create a new email post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--type]
+	 * : Required. Email type for the email (should be unique identifier, sanitized like a post slug).
+	 *
+	 * [--type_description]
+	 * : Email type description.
+	 *
+	 * [--subject]
+	 * : Required. Email subject line. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [--content]
+	 * : Required. Email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [--plain_text_content]
+	 * : Plain-text email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp email create --type=new-event --type_description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --content="<a href='{{{some.custom-token-url}}}'></a>A new event</a> was created" --plain_text_content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		// 'type' is required.
+		if ( empty( $assoc_args['type'] ) ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "The 'type' field must be filled in." );
+			return;
```

Comment:
The `error` method already bails/stops the execution. This `return` and the others you introduced below after the method are not necessary.